### PR TITLE
support urlRoot

### PR DESCRIPTION
**Pull request by GavinJoyce**
Fixes https://github.com/intercom/ember-href-to/issues/66

/cc @paddyobrien  this fixes a regression which was introduced with `1.10.0` where the rootURL wasn't being removed from the url before handling.

### CODE REVIEW

File: addon/href-to.js [hunk 1]
```diff
@@ -26,9 +26,10 @@ export default class {
 
   handle() {
     let router = this._getRouter();
+    let urlWithoutRoot = this.getUrlWithRoot();
```
Comment (paddyobrien) [inline addon/href-to.js]: Is this withRoot or withoutRoot? variable says one thing method name says another

Comment (GavinJoyce) [inline addon/href-to.js]: Thanks, it should be `getUrlWithoutRoot`

Comment (GavinJoyce) [inline addon/href-to.js]: Updated

File: addon/href-to.js [hunk 2]
```diff
@@ -66,14 +67,20 @@ export default class {
...
+      didRecognize = isInternal && router.router.recognizer.recognize(urlWithoutRoot);
     }
 
     return didRecognize;
   }
 
+  getUrlWithoutRoot() {
+    let url = this.url;
+    let rootUrl = this._getRootUrl();
+    return url.substr(rootUrl.length - 1);
```
Comment (paddyobrien) [inline addon/href-to.js]: Is it possible for rootUrl.length to `==` 0?

Comment (GavinJoyce) [inline addon/href-to.js]: The rootUrl needs to end in a `/`, an empty string isn't valid

Comment (GavinJoyce) [inline addon/href-to.js]: https://github.com/emberjs/ember.js/blob/v2.10.0/packages/ember-routing/lib/system/router.js#L89

Comment (GavinJoyce) [inline addon/href-to.js]: screen shot 2016-12-08 at 13 44 31

Comment (paddyobrien) [inline addon/href-to.js]: 👍

**Pull request merged**